etest/profile.py

```diff
@@ -0,0 +1,124 @@
+"""Profile definitions and builders."""
+
+import logging
+from enum import Enum
+
+
+class _arm_mappings(Enum):
```

Comment:
We can name this much better.  Right now it doesn't indicate what is mapped.  If this were a function how would you name it?

Reply:
What about `_arm_name_mappings`?

Reply:
It still doesn't tell me the story.  What is the domain and codomain of this mapping?  That might be a better place to start.

Reply:
The domain is a simplified identifier for the arm revision. The codomain is the full description of the revision to be fetched from the stage3 repos.

Reply:
Ok, so I'd call that canonicalise_identifier then.  As an Enum I'd call it CanonicalARMIdentifiers

Reply:
Actually, on second thought, calling it ARMIdentifier makes the most sense.  Maybe special aliases for the name and value to indicate the semantics (not required on that latter bit though).